Refactor ContactsPicker to show a clean search bar

guillett opened the pull request:
### Contributor checklist
<!-- replace the empty checkboxes [ ] below with checked ones [x] accordingly -->
- [x] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [x] My commits are rebased on the latest master branch
- [x] My commits are in nice logical chunks
- [x] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
 * iPhone Version: iPhone5,2, iOS Version: 9.3.5
 * iPhone Simulator: iPhone 6 iOS Version: 10
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message


### Description

* Replace UITableviewController by UIViewController
* Create a custom xib file

Closes #1527 and fixes #1516 

![capture d ecran 2016-12-12 a 14 14 06](https://cloud.githubusercontent.com/assets/1410356/21100690/c1eb7f14-c075-11e6-8396-e716e76f4932.png)
![device](https://cloud.githubusercontent.com/assets/1410356/21100689/c1eb0002-c075-11e6-8b7a-86a749489c9f.png)


Note:
* A string has to be localized (cf. device screenshot)
* I can't have a bundle that has avatars in the Invite table view whatever git revision I use (I used OWS master, 2.6.7.0 tag and after my commit)

michaelkirk commented on Signal/src/view controllers/ContactsPicker.swift:
```diff
@@ -307,25 +294,29 @@ open class ContactsPicker: UITableViewController, UISearchResultsUpdating, UISea
     
     // MARK: - Search Actions
+
+    open func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
```
This seems unused?

michaelkirk commented on Signal/src/view controllers/ContactsPicker.swift:
```diff
@@ -66,8 +68,8 @@ open class ContactsPicker: UITableViewController, UISearchResultsUpdating, UISea
-        // Don't obscure table header (search bar) with table index
-        tableView.sectionIndexBackgroundColor = UIColor.clear
+        searchBar.placeholder = NSLocalizedString("INVITE_FRIENDS_PICKER_SEARCHBAR_PLACEHOLDER", comment: "Search")
+        self.edgesForExtendedLayout = []
```
It's not obvious what this `edgesForExtendedLayout` is doing. Can you comment the code?

michaelkirk commented on Signal/src/view controllers/ContactsPicker.swift:
```diff
@@ -78,28 +80,15 @@ open class ContactsPicker: UITableViewController, UISearchResultsUpdating, UISea
         reloadContacts()
-        initializeSearchBar()
+        updateSearchResults(searchText: "")
+        self.tableView.reloadData()
```
This `reloadData` is redundant because we're doing it at the end of updateSearchResults.

michaelkirk commented on Signal/src/view controllers/ContactsPicker.swift:
```diff
@@ -307,25 +294,29 @@ open class ContactsPicker: UITableViewController, UISearchResultsUpdating, UISea
+        self.tableView.reloadData()
     }
     
     open func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
```
I no longer see a way to cancel the search.

Previously:
![search-filter](https://cloud.githubusercontent.com/assets/217057/21268622/286aead4-c375-11e6-81fc-982a1f0df2fd.jpg)

guillett commented:
Thanks a lot for your feedback. All your comments were 100% relevant. I removed the code that was no longer used and rebased the commit on the new master.

michaelkirk commented:
In https://github.com/WhisperSystems/Signal-iOS/commit/d7b27a4021dc52de4514de5d7b40e48cabd75d11, thanks!
